core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -132,6 +138,60 @@ public ClientFactoryBuilder workerGroup(EventLoopGroup workerGroup, boolean shut
         return this;
     }
 
+    /**
+     * Sets the factory that creates an {@link EventLoopScheduler} which is responsible for assigning an
+     * {@link EventLoop} to handle a connection to the specified {@link Endpoint}.
+     */
+    public ClientFactoryBuilder eventLoopSchedulerFactory(
+            Function<? super EventLoopGroup, ? extends EventLoopScheduler> eventLoopSchedulerFactory) {
+        checkState(maxEventLoopPerEndpoint == 0 && maxEventLoops.isEmpty(),
+                   "Cannot set eventLoopSchedulerFactory when maxEventLoop per endpoint is specified.");
+        this.eventLoopSchedulerFactory = requireNonNull(eventLoopSchedulerFactory, "eventLoopSchedulerFactory");
+        return this;
+    }
+
+    /**
+     * Sets the maximum number of event loops which will be used to handle connections to all
+     * {@link Endpoint}s except the ones specified by {@link #maxEventLoop(Endpoint, int)} and
+     * {@link #maxEventLoop(String, int)}.
+     */
+    public ClientFactoryBuilder maxEventLoopPerEndpoint(int maxEventLoopPerEndpoint) {
+        checkArgument(maxEventLoopPerEndpoint > 0,
+                      "maxEventLoopPerEndpoint: %s (expected: > 0)", maxEventLoopPerEndpoint);
+        checkState(eventLoopSchedulerFactory == null,
+                   "maxEventLoopPerEndpoint() and eventLoopSchedulerFactory() are mutually exclusive.");
+        this.maxEventLoopPerEndpoint = maxEventLoopPerEndpoint;
+        return this;
+    }
+
+    /**
+     * Sets the maximum number of event loops which will be used to handle connections to the specified
+     * {@link Endpoint}.
+     * This method is a shortcut of:
+     * <pre>{@code
+     * String authority = endpoint.authority();
+     * maxEventLoop(authority, maxEventLoop);
+     * }</pre>
+     */
+    public ClientFactoryBuilder maxEventLoop(Endpoint endpoint, int maxEventLoop) {
```

Comment:
`maxNumEventLoopsPerEndpoint`? I somehow feel more natural to have `PerEndpoint`..

Reply:
Yeah, that sounds more appropriate. Thanks!